### PR TITLE
Error handling in rest (JSON formatted errors) #019

### PR DESCRIPTION
**Pull request by Salim-54**
#### What does this PR do?

- Handles the errors in production

#### Description of Task to be completed? 

As a user, I should be able to 
- Get an error response in JSON format

#### How should this be manually tested?

- Clone the repo

- checkout to branch ``ft-error-handling-in-rest-#019``

- Run `npm install` to install modules

- Run `npm  run start` to start the development server

- In your API testing tool (postman recommended) call api `localhost:5000//api/v1/users` to add a todo task

#### Any background context you want to provide?

- Note that you should send the request with a valid FALSE endpoints to notice the error

#### What are the relevant Pivotal Tracker/trello stories?

- [link to my pivotal tracker/tello/jira ... implemented story](https://trello.com/c/aeDPLhYi/41-error-handling019)

#### Screenshots (if appropriate)
- N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by Xldivin — APPROVED** [review]

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]
Place `.env` file in `.gitignore` so that It's not committed and pushed. Actually, we don't push our `.env` files

File: src/controllers/blog.controller.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { blog } from 'express';
+import Blog from '../database/model/blog.model';
+
+exports.saveBlog  = async (req, res) => {
+
+    try{
+
+        const newBlog = await Blog.create(req.body);
+        // await newBlog.save();
+        res.status(201).json({
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:10]: remove the commented line 9, not good to push comments to master

File: src/controllers/blog.controller.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    try{
+        
+        const blog = await Blog.findById(req.params.id);
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                blog
+            }
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:46]: Uncomment that line as well. cz you need to return `404` when the blog doesn't exists

File: src/controllers/blog.controller.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            new: true,
+            runValidators: true
+        })
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                blog
+            }
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:68]: That line is needed but should be like on line 56. Youn needs to check if the blog exists before anything.
You'll first check if it exists from db

File: src/controllers/blog.controller.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+exports.deleteBlogById = async (req, res) => {
+    try{
+        await Blog.findByIdAndDelete(req.params.id);
+
+        res.status(204).json({
+            status: 'success', 
+            data: null
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:86]: That line is needed but should be like on line 79 Youn needs to check if the blog exists before anything.
You'll first check if it exists from db

File: src/controllers/query.controller.js [hunk 5]
```diff
@@ -1,15 +1,55 @@
...
-    const newQuery = new Query(query);
-    await newQuery.save();
-    res.status(201).json({success: true, data: newQuery});
-}
+exports.saveQuery  = async (req, res) => {
+
+    try{
+
+        const newQuery = await Query.create(req.body);
+        // await newQuery.save();
```
Comment (thepatrickniyo) [inline src/controllers/query.controller.js:9]: remove the commented line 9, not good to push comments to master

File: src/controllers/query.controller.js [hunk 6]
```diff
@@ -1,15 +1,55 @@
...
+    try{
+        
+        const query = await Query.findById(req.params.id);
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                query
+            }
+        });
+        // if(!query) return res.status(404).json({success: false, message: "Query not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/query.controller.js:46]: That line is needed but should be like on line 40 Youn needs to check if the blog exists before anything.

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]

**Pull request merged**